src/proxy_app/rate_limiter.py

```
@@ -348,6 +336,29 @@ async def can_use_provider(
+                    prov_name, model
+                )
+                if is_limited and retry_after_ms and retry_after_ms > 0:
+                    restored_until = now + (retry_after_ms / 1000.0)
+                    status.rate_limited_until = restored_until
+                    status.block_reason = REASON_RATE_LIMIT
+                    logger.info(
+                        f"Restored persisted rate-limit block for {key} "
+                        f"(retry_after={retry_after_ms}ms)"
+                    )
+                    return False, REASON_RATE_LIMIT
+            except Exception as e:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)